Mathlib/LinearAlgebra/RootSystem/WeylGroup.lean

```diff
@@ -0,0 +1,141 @@
+
+lemma range_weylGroup_weightHom :
+    MonoidHom.range ((Equiv.weightHom P).restrict P.weylGroup) =
+      Subgroup.closure (range P.reflection) := by
+  refine (Subgroup.closure_eq_of_le _ ?_ ?_).symm
+  · rintro - ⟨i, rfl⟩
+    simp only [MonoidHom.restrict_range, Subgroup.coe_map, Equiv.weightHom_apply, mem_image,
+      SetLike.mem_coe]
+    use Equiv.reflection P i
+    exact ⟨reflection_mem_weylGroup P i, Equiv.reflection_weightEquiv P i⟩
+  · rintro fg ⟨⟨w, hw⟩, rfl⟩
+    induction hw using Subgroup.closure_induction'' with
+    | one =>
+      change ((Equiv.weightHom P).restrict P.weylGroup) 1 ∈ _
+      simpa only [map_one] using Subgroup.one_mem _
+    | mem w' hw' =>
+      obtain ⟨i, rfl⟩ := hw'
+      simp only [MonoidHom.restrict_apply, Equiv.weightHom_apply, Equiv.reflection_weightEquiv]
+      simpa only [reflection_mem_weylGroup] using Subgroup.subset_closure (mem_range_self i)
+    | inv_mem w' hw' =>
+      obtain ⟨i, rfl⟩ := hw'
+      simp only [Equiv.reflection_inv, MonoidHom.restrict_apply, Equiv.weightHom_apply,
+        Equiv.reflection_weightEquiv]
+      simpa only [reflection_mem_weylGroup] using Subgroup.subset_closure (mem_range_self i)
+    | mul w₁ w₂ hw₁ hw₂ h₁ h₂ =>
+      simpa only [← Submonoid.mk_mul_mk _ w₁ w₂ hw₁ hw₂, map_mul] using Subgroup.mul_mem _ h₁ h₂
+
+lemma range_weylGroup_coweightHom :
+    MonoidHom.range ((Equiv.coweightHom P).restrict P.weylGroup) =
+      Subgroup.closure (range (MulOpposite.op ∘ P.coreflection)) := by
+  refine (Subgroup.closure_eq_of_le _ ?_ ?_).symm
+  · rintro - ⟨i, rfl⟩
+    simp only [MonoidHom.restrict_range, Subgroup.coe_map, Equiv.weightHom_apply, mem_image,
+      SetLike.mem_coe]
+    use Equiv.reflection P i
+    refine ⟨reflection_mem_weylGroup P i, by simp⟩
+  · rintro fg ⟨⟨w, hw⟩, rfl⟩
+    induction hw using Subgroup.closure_induction'' with
+    | one =>
+      change ((Equiv.coweightHom P).restrict P.weylGroup) 1 ∈ _
+      simpa only [map_one] using Subgroup.one_mem _
+    | mem w' hw' =>
+      obtain ⟨i, rfl⟩ := hw'
+      simp only [MonoidHom.restrict_apply, Equiv.coweightHom_apply, Equiv.reflection_coweightEquiv]
+      simpa only [reflection_mem_weylGroup] using Subgroup.subset_closure (mem_range_self i)
+    | inv_mem w' hw' =>
+      obtain ⟨i, rfl⟩ := hw'
+      simp only [Equiv.reflection_inv, MonoidHom.restrict_apply, Equiv.coweightHom_apply,
+        Equiv.reflection_coweightEquiv]
+      simpa only [reflection_mem_weylGroup] using Subgroup.subset_closure (mem_range_self i)
+    | mul w₁ w₂ hw₁ hw₂ h₁ h₂ =>
+      simpa only [← Submonoid.mk_mul_mk _ w₁ w₂ hw₁ hw₂, map_mul] using Subgroup.mul_mem _ h₁ h₂
+
+/-- The permutation representation of the Weyl group induced by `reflection_perm`. -/
+abbrev weylGroupToPerm := ((Equiv.indexHom P).restrict P.weylGroup)
+
+@[simp]
+lemma weylGroupToPerm_apply_reflection : --deprecate?
+    P.weylGroupToPerm ⟨Equiv.reflection P i, reflection_mem_weylGroup P i⟩ =
+      P.reflection_perm i :=
+  rfl
+
```

Comment:
I think we're safe just to drop this: I think it's still too soon for there to be dependencies on this material.
```suggestion
```